### PR TITLE
RUN-4205 transitioned window.js to TypeScript

### PR DESCRIPTION
**Pull request by whyn07m3**
ℹ️ This PR transitions api_handler's `window.js` to TypeScript

**Automated test results**
✅ [Window 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b212e2d54b21953031f35c4)
✅ [Window 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b212f3554b21953031f35c5)

### CODE REVIEW

**Comment by nicholasdgoodman** [comment]
Any reason GitHub is showing this as a delete / add instead of a rename and changed?

**Comment by whyn07m3** [comment]
@nicholasdgoodman because almost every line in that file has changed ¯\_(ツ)_/¯

File: src/browser/api_protocol/api_handlers/window.ts [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+    const { code } = payload;
+    const dataAck = Object.assign({}, successAck);
+    const windowIdentity = getTargetWindowIdentity(payload);
+    let { uuid: pUuid } = windowIdentity;
+
+    while (pUuid) {
+        if (pUuid === identity.uuid) {
+            return Window.executeJavascript(windowIdentity, code, (err: Error, result: any) => {
+                if (err) {
+                    nack(err); // TODO: this nack doesn't follow the protocol
```
Comment (wenjunche) [inline src/browser/api_protocol/api_handlers/window.ts]: Nacker = (payload: APIPayloadNack | Error) ?

Comment (whyn07m3) [reply src/browser/api_protocol/api_handlers/window.ts]: @datamadic @rdepena

**Comment by datamadic** [comment]
@whyn07m3 can you re-post and this will get merged

**Comment by whyn07m3** [comment]
Automated tests after rebasing:
✅ [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b34f04a54b21953031f3647)
✅ [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b34f16354b21953031f364a)

**Pull request merged**